typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -203,36 +208,105 @@ export class EvmERC20WarpRouteReader extends EvmRouterReader {
+    const contractVerificationStatus: Record<
+      string,
+      ContractVerificationStatus
+    > = {};
 
     const contractType = (await isProxy(this.provider, address))
       ? VerifyContractTypes.Proxy
       : VerifyContractTypes.Implementation;
 
-    virtualConfig.contractVerificationStatus[contractType] =
+    if (this.multiProvider.isLocalRpc(chain)) {
+      this.logger.debug('Skipping verification for local endpoints');
+      return { [contractType]: ContractVerificationStatus.Skipped };
+    }
+    contractVerificationStatus[contractType] =
       await this.contractVerifier.getContractVerificationStatus(chain, address);
 
     if (contractType === VerifyContractTypes.Proxy) {
-      virtualConfig.contractVerificationStatus.Implementation =
+      contractVerificationStatus[VerifyContractTypes.Implementation] =
         await this.contractVerifier.getContractVerificationStatus(
           chain,
           await proxyImplementation(this.provider, address),
         );
 
       // Derive ProxyAdmin status
-      virtualConfig.contractVerificationStatus.ProxyAdmin =
+      contractVerificationStatus[VerifyContractTypes.ProxyAdmin] =
         await this.contractVerifier.getContractVerificationStatus(
           chain,
           await proxyAdmin(this.provider, address),
         );
     }
+    return contractVerificationStatus;
+  }
+
+  async getOwnerStatus(chain: ChainName, address: Address) {
+    let ownerStatus: Record<string, OwnerStatus> = {};
+    if (this.multiProvider.isLocalRpc(chain)) {
+      this.logger.debug('Skipping owner verification for local endpoints');
+      return {
+        [address]: OwnerStatus.Skipped,
+      };
+    }
+
+    const provider = this.multiProvider.getProvider(chain);
+    const owner = await Ownable__factory.connect(address, provider).owner();
+
+    ownerStatus[owner] = (await isAddressActive(provider, owner))
+      ? OwnerStatus.Active
+      : OwnerStatus.Inactive;
+
+    // Heuristically check if the owner could be a safe by calling an expected function
+    // This status will overwrite 'active' status
+    try {
+      await IOwnerManager__factory.connect(owner, provider).getThreshold();
+      ownerStatus[owner] = OwnerStatus.GnosisSafe;
+    } catch {
+      this.logger.debug(`${owner} is not a safe`);
+    }
+
+    // Check Proxy admin and implementation recursively
+    const contractType = (await isProxy(this.provider, address))
+      ? VerifyContractTypes.Proxy
+      : VerifyContractTypes.Implementation;
+    if (contractType === VerifyContractTypes.Proxy) {
+      // Implementation
+      ownerStatus = {
+        ...ownerStatus,
+        ...(await this.getOwnerStatus(
+          chain,
+          await proxyImplementation(this.provider, address),
+        )),
+      };
+
+      // ProxyAdmin
+      ownerStatus = {
+        ...ownerStatus,
+        ...(await this.getOwnerStatus(
+          chain,
+          await proxyAdmin(provider, address),
+        )),
+      };
```

Comment:
Why not promise.all and then add both results to the `ownerStatus`  obj

Reply:
good idea